File: python/linkedList.py

class Node:

    def __init__(self, data):
        self.data = data
        self.nxt = None
    
class LinkedList:

    def __init__(self):
        self.head = None
    
    def push(self, data):
        new_node = Node(data)
        new_node.nxt = self.head
        self.head = new_node
    
    def insert(self, prev, item):
        if prev is None:
            print("the previous node must be in thelinked list")
        new_node = Node(item)
        new_node.nxt = prev.nxt
        prev.nxt = new_node
    def append(self, item):
        new_node = Node(item)
        if self.head is None:
            self.head = new_node
        
        curr = self.head
        while(curr.nxt):
            curr = curr.nxt
        curr.nxt = new_node
    def print(self):
        curr = self.head
        while (curr):
            print(curr.data)
            curr = curr.nxt


l = LinkedList()
l.push(2)
l.push(3)
l.insert(l.head, 4)
l.append(5)
l.print()

File: go/linkedlist.go
package main

import "fmt"

func main() {
	fmt.Println()
	l1 := new(LinkedList)
	fmt.Println(l1)
	l1.PushToHead(2)
	l1.PushToHead(3)
	l1.PushToHead(4)
	l1.Append(5)
	l1.Print()
}

type node struct {
	data interface{}
	next *node
}

type LinkedList struct {
	head *node
}

func (l *LinkedList) PushToHead(val interface{}) {
	if l.head == nil {
		n := &node{data: val, next: l.head}
		l.head = n
	} else {
		new_node := new(node)
		new_node = l.head
		n2 := &node{data: val, next: new_node}
		l.head = n2

	}
}
func (l *LinkedList) Append(val interface{}) {
	if l.head == nil {
		l.head = &node{data: val, next: nil}
	}
	curr := l.head
	for curr.next != nil {
		curr = curr.next
	}
	curr.next = &node{data: val, next: nil}
}

func (l *LinkedList) Print() {
	curr := l.head
	for curr != nil {
		fmt.Println(curr.data)
		curr = curr.next
	}
}
